fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/BinaryObjectTest.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import static org.mockito.Matchers.eq;
+
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Resource;
+
+import edu.unc.lib.dl.fcrepo4.RepositoryPathConstants;
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Ebucore;
+import edu.unc.lib.dl.rdf.Fcrepo4Repository;
+import edu.unc.lib.dl.rdf.Premis;
+
+public class BinaryObjectTest extends AbstractFedoraTest {
+	@Mock
+	private PID mockPid;
+	
+	private BinaryObject binObj;
+	private Model model;
+	private Resource resource;
+	
+	private ByteArrayInputStream stream;
+	
+	private static final String BASE_PATH = FEDORA_BASE;
+	private static final String METADATA_PATH = RepositoryPathConstants.FCR_METADATA;
+	
+	@Before
+	public void init() throws URISyntaxException {
+		MockitoAnnotations.initMocks(this);
+		
+		byte[] buf = new byte[10];
+		stream = new ByteArrayInputStream(buf);
+		
+		when(mockPid.getRepositoryUri()).thenReturn(new URI(BASE_PATH));
+		binObj = new BinaryObject(mockPid, repository, dataLoader);
+		
+		when(dataLoader.loadModel(binObj)).thenReturn(dataLoader);
+		
+		setupModel();
+	}
+	
+	private void setupModel() {
+		model = ModelFactory.createDefaultModel();
+		binObj.storeModel(model);
+		resource = binObj.getResource();
+	}
+
+	@Test
+	public void testGetMetadataUri() {
+		assertEquals(BASE_PATH + METADATA_PATH, binObj.getMetadataUri().toString());
+	}
+
+	@Test
+	public void testValidateType() {
+		// Return the correct RDF types
+		 	List<String> types = Arrays.asList(Fcrepo4Repository.Binary.toString());
```

Comment:
Sorry, somehow my spacing got messed up with the copy/paste I did earlier. Should look bettter now.